packages/strapi-admin/admin/src/containers/Webhooks/EditView/tests/reducer.test.js

```diff
@@ -0,0 +1,231 @@
+import { fromJS } from 'immutable';
+import reducer from '../reducer';
+
+describe('Admin | containers | Webhooks | EditView | reducer', () => {
+  const header = { key: '', value: '' };
+
+  const initialWebhook = {
+    events: [],
+    headers: [header],
+    name: '',
+    url: '',
+  };
+
+  const initialState = fromJS({
+    formErrors: {},
+    initialData: initialWebhook,
+    isTriggering: false,
+    modifiedData: initialWebhook,
+    triggerResponse: {},
+  });
+
+  it('It should add new header to modifiedData on ADD_NEW_HEADER', () => {
+    const state = initialState;
+
+    const action = {
+      type: 'ADD_NEW_HEADER',
+      keys: ['headers'],
+    };
+
+    const updatedHeaders = state
+      .getIn(['modifiedData', 'headers'])
+      .push(fromJS(header));
+
+    const expectedState = state.setIn(
+      ['modifiedData', ...action.keys],
+      updatedHeaders
+    );
+
+    expect(reducer(state, action)).toEqual(expectedState);
+  });
+
+  it('It should format headers as an array of an empty object if received headers is empty and update initialData and modifiedData states with received data on GET_DATA_SUCCEEDED', () => {
+    const state = initialState;
+
+    const receivedData = {
+      id: 1,
+      name: 'webhook 1',
+      url: 'http://localhost:5000',
+      headers: {},
+      events: ['entry.create', 'entry.update', 'entry.delete'],
+      isEnabled: true,
+    };
+
+    const action = {
+      type: 'GET_DATA_SUCCEEDED',
+      data: receivedData,
+    };
+
+    const formattedHeaders = [header];
+    const data = fromJS(receivedData).update('headers', () =>
+      fromJS(formattedHeaders)
+    );
+    const expectedState = state
+      .update('initialData', () => data)
+      .update('modifiedData', () => data);
+
+    expect(reducer(state, action)).toEqual(expectedState);
+  });
+
+  it('It should format headers as an array of objects if received headers is not empty and update initialData and modifiedData states with received data on GET_DATA_SUCCEEDED', () => {
+    const state = initialState;
+
+    const receivedData = {
+      id: 1,
+      name: 'webhook 1',
+      url: 'http://localhost:5000',
+      headers: {
+        accept: 'text/html',
+        authorization: 'Basic YWxhZGRpbjpvcGVuc2VzYW1l',
+      },
+      events: ['entry.create', 'entry.update', 'entry.delete'],
+      isEnabled: true,
+    };
+
+    const action = {
+      type: 'GET_DATA_SUCCEEDED',
+      data: receivedData,
+    };
+
+    const headers = receivedData.headers;
+    const formattedHeaders = Object.keys(headers).map(key => {
```

Comment:
Same here it is best to avoid logic in your tests and prefer "hardcoding" your results.